### PR TITLE
[dhctl][node-manager] Old dockerCfg format support (construct auth from username and password)

### LINKED ISSUE #194: [bashible] install registry packages fails, if registryDockerCFG in old format
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.24

### Expected Behavior

if registryDockerCfg don't contain "auth" field, bashible-apiserver must use username and password fields to  construct registry.auth parameter for bashible.

### Actual Behavior

if registryDockerCfg don't contain "auth" field, bashible-apiserver cannot parse this record properly, and bashible don't get proper formed registry.auth parameter.

### Steps To Reproduce

Run cluster with registryDockerCfg without auth field.

### Additional Information

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by name212**
## Description
Close #194

Add support in dhctl and bashible-apiserver.

if registryDockerCfg does not contain "auth" field, use username and password fields to construct registry.auth parameter for bashible.

## Why do we need it, and what problem does it solve?
We should support old docker config format.
We can not bootstrap cluster and bashible does not work, if we change deckhouse-registry secret in old format. 

## Changelog entries

```changes
section: dhctl
type: fix
summary: Old dockerCfg format support (construct auth from username and password)
impact_level: low
---
section: node-manager
type: fix
summary: Old dockerCfg format support in bashible-apiserver (construct auth from username and password)
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/context_builder.go [hunk 1]
```diff
@@ -495,7 +502,9 @@ type registryInputData struct {
 
 type dockerCfg struct {
 	Auths map[string]struct {
-		Auth string `json:"auth"`
+		Auth     *string `json:"auth"`
+		Username *string `json:"username"`
+		Password *string `json:"password"`
```
Comment (yalosev) [inline modules/040-node-manager/images/bashible-apiserver/pkg/template/context_builder.go:507]: Honestly, I don't see the need of pointers here. You don't need 3 states, only fallback. You can always compare with an empty string and fallback if true. It's not critical, so you can hold this but this gives us extra gc workload without a reason.

We can wait for @RomanenkoDenys opinion

File: dhctl/pkg/config/meta.go [hunk 2]
```diff
@@ -514,14 +516,22 @@ func (m *MetaConfig) ParseRegistryData() (map[string]interface{}, error) {
 			return nil, fmt.Errorf("cannot base64 decode docker cfg: %v", err)
 		}
 
-		log.DebugF("parse registry data: dockerCfg after base64 decode = %s\n", bytes)
+		log.DebugF("parse registry data: dockerCfg after base64 decode = %s\n", string(bytes))
```
Comment (RomanenkoDenys) [inline dhctl/pkg/config/meta.go:519]: in the internal logging functions uses fmt.Sprintf, so we don't need to do conversion.
```
func simpleFormat(format string, a ...interface{}) string {
	var msg string
	if len(a) > 0 {
		msg = fmt.Sprintf(format, a...)
	} else {
		msg = fmt.Sprintf("%s", format)
	}

	return msg
}
```

File: dhctl/pkg/config/meta_test.go [hunk 3]
```diff
@@ -131,21 +138,59 @@ provider:
...
+	return base64.StdEncoding.EncodeToString([]byte(auth))
 }
 
-func TestPrepare(t *testing.T) {
-	t.Run("Registry", func(t *testing.T) {
-		dataToRender := map[string]interface{}{
-			"dockerCfg":  generateDockerCfg(),
-			"imagesRepo": "r.example.com/deckhouse/ce/",
+func generateDockerCfg(host, username, password string) string {
+	return fmt.Sprintf(`{ "auths": { "%s": { "auth": "%s" } } }`, host, dockerCfgAuth(username, password))
```
Comment (RomanenkoDenys) [inline dhctl/pkg/config/meta_test.go:147]:
```suggestion
	return fmt.Sprintf(`{"auths":{"%s":{"auth":"%s"}}}`, host, dockerCfgAuth(username, password))
```

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/context_builder.go (hunk 1 above)
Comment (RomanenkoDenys) [reply modules/040-node-manager/images/bashible-apiserver/pkg/template/context_builder.go:507]: I think this is not important, but this struct is copy of struct from [dhctl/pkg/config/meta.go](https://github.com/deckhouse/deckhouse/pull/1515/files#diff-4b6226a078c0950de166cd15009acfea9b56a97a8264f43eca39cc72b2703c91), so we can leave this as is or change it in two places.

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/context_builder.go (hunk 1 above)
Comment (name212) [reply modules/040-node-manager/images/bashible-apiserver/pkg/template/context_builder.go]: Ok. I checked that empty string is normal case for checking existing auth string. I removed pointers.

File: dhctl/pkg/config/meta.go (hunk 2 above)
Comment (name212) [reply dhctl/pkg/config/meta.go]: Oh, Thank you. I removed it.

**Review by RomanenkoDenys — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Pull request merged**